keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/PromotionRepository.kt

```diff
@@ -0,0 +1,20 @@
+package com.netflix.spinnaker.keel.persistence
+
+import com.netflix.spinnaker.keel.api.DeliveryArtifact
+import com.netflix.spinnaker.keel.api.DeliveryConfig
+
+interface PromotionRepository {
```

Comment:
The idea here is that we can easily record what version of a particular artifact is currently in a particular environment. In theory we could also do this via CloudDriver but then we'd need to look at a particular resource. Still debating with myself whether that would actually be better.

Reply:
re: comment above, I kind of like us tracking the states independently of what clouddriver says is currently there - gives us opportunities to also track whether an environment went into a bad state due to a change and whether an artifact was rolled back (or fixed foward possiblity). 

Not sure we could get that level of metadata on an artifact's lifecycle from clouddriver without essentially building a repository over there